src/ripple/app/ledger/impl/MetaView.cpp

```diff
@@ -543,11 +543,12 @@ Json::Value MetaView::getJson (int) const
 
 void
 MetaView::apply (BasicView& to,
-    STTx const& tx, TER ter, beast::Journal j)
+    std::shared_ptr<STTx const> const& tx,
```

Comment:
The queue needs to store shared_ptrs so that objects don't expire. Did I
get overzealous?
On Jul 1, 2015 5:02 PM, "Vinnie Falco" <notifications@github.com> wrote:

> In src/ripple/app/ledger/impl/MetaView.cpp
> <https://github.com/ripple/rippled/pull/1160#discussion_r33724118>:
>
> > @@ -543,11 +543,12 @@ Json::Value MetaView::getJson (int) const
> >
> >  void
> >  MetaView::apply (BasicView& to,
> > -    STTx const& tx, TER ter, beast::Journal j)
> > +    std::shared_ptr<STTx const> const& tx,
>
> I don't understand, what was the point of using std::shared_ptr?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ripple/rippled/pull/1160/files#r33724118>.
>